### PR TITLE
Register themes/gitea submodule

### PR DESCRIPTION
**Pull request by strk**
I think it helps, will leave decision to others

### CODE REVIEW

**Comment by andreynering** [comment]
Maybe subtree?

**Comment by strk** [comment]
Why subtree ? Submodule only drops a reference to the SHA and the repository public URL in the main repository. I don't know what subtree does.

**Comment by andreynering** [comment]
I was just thinking... I would be a kind of vendoring, a PR would need to be open every time one want to update the theme. Maybe submodules are better if we want it automatically updated. Let's @tboerger decide this.

File: .drone.yml [hunk 1]
```diff
@@ -6,7 +6,6 @@ pipeline:
     image: webhippie/hugo:latest
     commands:
       - apk -U add git
-      - git clone https://github.com/go-gitea/theme.git themes/gitea
```
Comment (andreynering) [inline .drone.yml:9]: I think this would need to be something like:

```
git submodule update
```

Comment (strk) [reply .drone.yml:9]: `git submodule update` would not do anything as submodules were just initialized.
I guess you wanted to instead `pull` the submodule to be on the tip of the master branch in its remote ?

It depends on what you want drone to do: use the theme approved in the master branch of `website` or use whatever happens to be in the master branch of `theme`

Comment (strk) [reply .drone.yml:9]: For the record: `git submodule foreach git pull origin master` is what you would use if you wanted to set all submodules to their master branch on the origin remote.

**Comment by tboerger** [comment]
Simply NO! This won't work with an automated workflow. I have already created PRs to add a make task and it's also documented. So here a BIG -1 from my side.

**Comment by strk** [comment]
On Thu, Nov 10, 2016 at 03:23:08AM -0800, Thomas Boerger wrote:

> Simply NO! This won't work with an automated workflow. I have already created PRs to add a make task and it's also documented. So here a BIG -1 from my side.

I suspected your veto.
Anyway "won't work with automated workflow" is not true...

**Comment by tboerger** [comment]
It won't work unless you allow the CI to push changes, and this is nothing I would even consider if I compare it with the current setup.

**Comment by strk** [comment]
@tboerger the current setup doesn't push any changes, just adds a 'git' command and clones a repository. Both thing that with the setup in the PR are not needed

**Comment by metalmatze** [comment]
A _No_ from me as well. 
Submodules need to be updated to update the theme, the current pipeline that was setup by @tboerger is way superior. We really don't need submodules. 
I suggest to add a `Makefile` and in `make all` it pulls everything to get one started locally.

**Comment by strk** [comment]
Ok then. Makefile was added by @tboerger, waiting for lgtm in  #3

**Pull request closed without merging**